dxp/layout_rendering.md

```diff
@@ -0,0 +1,173 @@
+# Layout (page) rendering in DXP
+
+This document explains how a request to render a page is serviced in DXP.
+
+## What gets requested when asking for http://localhost:8080?
+
+First, we are going to list what external and internal requests are fired whenever someone hits DXP's root page.
+
+> Note that, in addition to the requests a browser can make to our server, the server can dispatch internal requests to resources which undergo the standard request lifecycle, simulating an external request without any need to hit the HTTP transport layer at all. Usually the responses of these internal requests are included in the originating (external) request.
+
+> All this is done leveraging the [`RequestDispatcher` API](https://docs.oracle.com/javaee/7/api/javax/servlet/RequestDispatcher.html) that can be retrieved from any [`ServletRequest`](https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getRequestDispatcher-java.lang.String-). You can see an example of how it is used in [`MainServlet.java`](https://github.com/liferay/liferay-portal/blob/2448ae35c6a8232af153acb0f366eafad59f48c4/portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java#L1119).
+
+So, whenever someone hits http://localhost:8080, the following requests are serviced by Tomcat:
```

Comment:
Should you say?

> serviced by the application server (usually Tomcat):

Also, the word "serviced" is a bit strained to my ear. "served" or "handled" might be a bit more natural. (The "serviced" up in the first paragraph sounds a bit funny in the same way.)